pkg/endpoint/policy.go

```diff
@@ -189,15 +189,14 @@ func (e *Endpoint) regenerateConsumable(owner Owner, labelsMap *identityPkg.Iden
 	// L4 policy needs to be applied on two conditions
 	// 1. The L4 policy has changed
 	// 2. The set of applicable security identities has changed.
-	if e.RealizedL4Policy != c.L4Policy || e.LabelsMap != labelsMap {
-
-		desiredPolicyKeys = e.computeDesiredL4PolicyMapEntries(labelsMap, c.L4Policy)
+	if e.RealizedL4Policy != e.DesiredL4Policy || e.LabelsMap != labelsMap {
+		desiredPolicyKeys = e.computeDesiredL4PolicyMapEntries(labelsMap, e.DesiredL4Policy)
 		// Reuse the common policy, will be used in lxc_config.h (CFG_CIDRL4_INGRESS and CFG_CIDRL4_EGRESS)
-		e.RealizedL4Policy = c.L4Policy
+		//e.DesiredL4Policy = c.L4Policy
```

Comment:
Remove this line instead of commenting it out.

Reply:
Fixed.